SpeechResponder/ScriptResolver.cs

```diff
@@ -646,25 +645,32 @@ private BuiltinStore buildStore(Dictionary<string, Cottle.Value> vars)
             store["CommodityMarketDetails"] = new NativeFunction((values) =>
             {
                 CommodityMarketQuote result = null;
+                CommodityMarketQuote CommodityDetails(string commodityLocalizedName, Station station)
+                {
+                    return station?.commodities?.FirstOrDefault(c => c.localizedName == commodityLocalizedName);
+                }
+
                 if (values.Count == 1)
                 {
                     // Named commodity, current station
                     Station station = EDDI.Instance.CurrentStation;
-                    result = station == null ? null : station.commodities.FirstOrDefault(c => c.definition.localizedName == values[0].AsString);
+                    result = CommodityDetails(values[0].AsString, station);
                 }
                 else if (values.Count == 2)
                 {
                     // Named commodity, named station, current system 
                     StarSystem system = EDDI.Instance.CurrentStarSystem;
-                    Station station = system != null && system.stations != null ? system.stations.FirstOrDefault(v => v.name == values[1].AsString) : null;
-                    result = station == null ? null : station.commodities.FirstOrDefault(c => c.definition.localizedName == values[0].AsString);
+                    string stationName = values[1].AsString;
```

Comment:
In case you were wondering: this was done as a micro-optimisation, and to make the predicate as simple as possible. I don't know whether the optimiser is smart enough to do this for us.